code/datums/mood_events/drink_events.dm

```diff
@@ -1,16 +1,36 @@
 /datum/mood_event/drunk
 	mood_change = 3
 	description = "Everything just feels better after a drink or two."
+	/// The blush overlay to display when the owner is drunk
+	var/datum/bodypart_overlay/simple/emote/blush_overlay
+	/// The bodypart that the blush_overlay was applied to
+	var/datum/weakref/attached_bodypart
 
 /datum/mood_event/drunk/add_effects(param)
-	// Display blush visual
-	ADD_TRAIT(owner, TRAIT_BLUSHING, "[type]")
-	owner.update_body()
+	// Display blush overlay
+	if(!ishuman(owner))
```

Comment:
this code is *significantly* uglier than what was here before. can we make this all a simpler proc like `blush_overlay = owner.give_body_part_overlay(/datum/bodypart_overlay/simple/emote/blush, BODY_ZONE_HEAD)` (or even having BODY_ZONE_HEAD be a default created by blush), with the remove_effects being `QDEL_NULL(blush_overlay)`?